tests/src/test/scala/org/http4s/util/ProcessAppSpec.scala

```diff
@@ -0,0 +1,59 @@
+package org.http4s
+package util
+
+import scalaz.concurrent.Task
+import scalaz.stream.Process
+
+class ProcessAppSpec extends Http4sSpec {
+
+  "ProcessApp" should {
+    "Terminate Server on a Initial Process" in {
+      val myError = new Throwable("Bad Initial Process")
+      class TestApp extends ProcessApp {
+        override def main(args: List[String]): Process[Task, Unit] = {
+          Process.fail(myError)
+        }
+      }
+      new TestApp().main(Array.empty[String]) should_== (())
+    }
+
+    "Terminate Server on a Bad Second Value" in {
```

Comment:
I'm not sure what this is testing that the first test didn't.  The point is whether the Process fails or not, not that it emitted first.  But test until you're no longer scared, as they say...